### PR TITLE
fix nil pointer, add daemon and statefulsets

### PR DESCRIPTION
**Pull request by AjayTripathy**

### CODE REVIEW

File: costmodel/promparsers.go [hunk 1]
```diff
@@ -380,6 +380,41 @@ func GetStatefulsetMatchLabelsMetrics(queryResult interface{}, defaultClusterID
...
 }
 
+func GetPodDaemonsetsWithMetrics(queryResult interface{}, defaultClusterID string) (map[string]string, error) {
+	toReturn := make(map[string]string)
+	result, err := NewQueryResults(queryResult)
+	if err != nil {
+		return toReturn, err
+	}
+	for _, val := range result {
+		// We want Deployment, Namespace and ClusterID for key generation purposes
```
Comment (nikovacevic) [inline costmodel/promparsers.go]: This looks like an un-updated copy/paste comment

File: costmodel/promparsers.go [hunk 2]
```diff
@@ -380,6 +380,41 @@ func GetStatefulsetMatchLabelsMetrics(queryResult interface{}, defaultClusterID
...
+			return toReturn, err
+		}
+
+		clusterID, err := val.GetString("cluster_id")
+		if clusterID == "" {
+			clusterID = defaultClusterID
+		}
+
+		pod, err := val.GetString("pod")
+		if clusterID == "" {
```
Comment (nikovacevic) [inline costmodel/promparsers.go]: Should this be `pod == ""`?

File: costmodel/costmodel.go [hunk 3]
```diff
@@ -2144,6 +2166,16 @@ func (cm *CostModel) costDataRange(cli prometheusClient.Client, clientset kubern
 			}
 		}
 
+		var podStatefulSets []string
+		if _, ok := podStatefulsetsMapping[nsKey]; ok {
```
Comment (nikovacevic) [inline costmodel/costmodel.go]: If `nsKey := c.Namespace + "," + c.ClusterID` how does this match with the `podStatefulsetsMapping` keys, which are `nsKey := ns + "," + ss + "," + clusterID`?

Comment (AjayTripathy) [reply costmodel/costmodel.go]: podStatefulsetsMapping should be a map[string]map[string][]string where the key is two levels, [namespace,clusterid][podname]

Comment (nikovacevic) [reply costmodel/costmodel.go]: Yes, but I'm pointing out that there might be in issue here: https://github.com/kubecost/cost-model/blob/93f04c52cf43e32111747f30703a3e23769ea51f/costmodel/promparsers.go#L376

The type is still `map[string]map[string][]string`, but I think the top-level keys will not match.

Comment (nikovacevic) [reply costmodel/costmodel.go]: Perhaps that `nsKey` in the promparsers for StatefulSets should just be `ns,clusterId`, no?

Comment (nikovacevic) [reply costmodel/costmodel.go]: Or perhaps the function `getPodDeploymentsWithMetrics` is translating between the two. Looks like that might be the case. It's pretty confusing in there, but if it seems to be working then no need to clarify.

**Pull request merged**